### ISSUE TITLE
Runtime Error in stage "Sorting update streams"

### ISSUE THREAD

**Issue by yaobaiwei** [ldbc/ldbc_snb_datagen_spark#82]
When i launched the program with Hadoop, reported the following error log:
```
************************************************
* Sorting update streams  *
************************************************
19/05/06 12:32:59 INFO impl.TimelineClientImpl: Timeline service address: http://worker21:8188/ws/v1/timeline/
19/05/06 12:32:59 INFO client.RMProxy: Connecting to ResourceManager at /172.16.104.21:8032
19/05/06 12:32:59 WARN mapreduce.JobSubmitter: Hadoop command-line option parsing not performed. Implement the Tool interface and execute your application with ToolRunner to remedy this.
19/05/06 12:33:00 INFO mapreduce.JobSubmitter: Cleaning up the staging area /tmp/hadoop-yarn/staging/yanda/.staging/job_1556954960956_0089
Error during execution
Input path does not exist: hdfs://proj99:9000/user/yanda/hadoop/temp_updateStream_person_0_0
org.apache.hadoop.mapreduce.lib.input.InvalidInputException: Input path does not exist: hdfs://proj99:9000/user/yanda/hadoop/temp_updateStream_person_0_0
	at org.apache.hadoop.mapreduce.lib.input.FileInputFormat.singleThreadedListStatus(FileInputFormat.java:321)
	at org.apache.hadoop.mapreduce.lib.input.FileInputFormat.listStatus(FileInputFormat.java:264)
	at org.apache.hadoop.mapreduce.lib.input.SequenceFileInputFormat.listStatus(SequenceFileInputFormat.java:59)
	at org.apache.hadoop.mapreduce.lib.input.FileInputFormat.getSplits(FileInputFormat.java:385)
	at org.apache.hadoop.mapreduce.JobSubmitter.writeNewSplits(JobSubmitter.java:597)
	at org.apache.hadoop.mapreduce.JobSubmitter.writeSplits(JobSubmitter.java:614)
	at org.apache.hadoop.mapreduce.JobSubmitter.submitJobInternal(JobSubmitter.java:492)
	at org.apache.hadoop.mapreduce.Job$10.run(Job.java:1296)
	at org.apache.hadoop.mapreduce.Job$10.run(Job.java:1293)
	at java.security.AccessController.doPrivileged(Native Method)
	at javax.security.auth.Subject.doAs(Subject.java:422)
	at org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1628)
	at org.apache.hadoop.mapreduce.Job.submit(Job.java:1293)
	at org.apache.hadoop.mapreduce.Job.waitForCompletion(Job.java:1314)
	at ldbc.snb.datagen.hadoop.HadoopUpdateStreamSorterAndSerializer.run(HadoopUpdateStreamSorterAndSerializer.java:130)
	at ldbc.snb.datagen.generator.LDBCDatagen.runGenerateJob(LDBCDatagen.java:197)
	at ldbc.snb.datagen.generator.LDBCDatagen.main(LDBCDatagen.java:340)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at org.apache.hadoop.util.RunJar.run(RunJar.java:221)
	at org.apache.hadoop.util.RunJar.main(RunJar.java:136)
Exception in thread "main" org.apache.hadoop.mapreduce.lib.input.InvalidInputException: Input path does not exist: hdfs://proj99:9000/user/yanda/hadoop/temp_updateStream_person_0_0
	at org.apache.hadoop.mapreduce.lib.input.FileInputFormat.singleThreadedListStatus(FileInputFormat.java:321)
	at org.apache.hadoop.mapreduce.lib.input.FileInputFormat.listStatus(FileInputFormat.java:264)
	at org.apache.hadoop.mapreduce.lib.input.SequenceFileInputFormat.listStatus(SequenceFileInputFormat.java:59)
	at org.apache.hadoop.mapreduce.lib.input.FileInputFormat.getSplits(FileInputFormat.java:385)
	at org.apache.hadoop.mapreduce.JobSubmitter.writeNewSplits(JobSubmitter.java:597)
	at org.apache.hadoop.mapreduce.JobSubmitter.writeSplits(JobSubmitter.java:614)
	at org.apache.hadoop.mapreduce.JobSubmitter.submitJobInternal(JobSubmitter.java:492)
	at org.apache.hadoop.mapreduce.Job$10.run(Job.java:1296)
	at org.apache.hadoop.mapreduce.Job$10.run(Job.java:1293)
	at java.security.AccessController.doPrivileged(Native Method)
	at javax.security.auth.Subject.doAs(Subject.java:422)
	at org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1628)
	at org.apache.hadoop.mapreduce.Job.submit(Job.java:1293)
	at org.apache.hadoop.mapreduce.Job.waitForCompletion(Job.java:1314)
	at ldbc.snb.datagen.hadoop.HadoopUpdateStreamSorterAndSerializer.run(HadoopUpdateStreamSorterAndSerializer.java:130)
	at ldbc.snb.datagen.generator.LDBCDatagen.runGenerateJob(LDBCDatagen.java:197)
	at ldbc.snb.datagen.generator.LDBCDatagen.main(LDBCDatagen.java:340)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at org.apache.hadoop.util.RunJar.run(RunJar.java:221)
	at org.apache.hadoop.util.RunJar.main(RunJar.java:136)
```

I tracked into the code, and found that where the probelm is, starting from Line 185 in  LDBCDataGen.java:
```
List<String> personStreamsFileNames = new ArrayList<String>(); 
            List<String> forumStreamsFileNames = new ArrayList<String>(); 
            for (int i = 0; i < DatagenParams.numThreads; ++i) { 
                int numPartitions = conf.getInt("ldbc.snb.datagen.serializer.numUpdatePartitions", 1); 
                for (int j = 0; j < numPartitions; ++j) { 
                    personStreamsFileNames.add(DatagenParams.hadoopDir +  "/temp_updateStream_person_" + i + "_" + j); 
                    if (conf.getBoolean("ldbc.snb.datagen.generator.activity", false)) { 
                        forumStreamsFileNames.add(DatagenParams.hadoopDir +  "/temp_updateStream_forum_" + i + "_" + j); 
                    } 
                } 
            } 
HadoopUpdateStreamSorterAndSerializer updateSorterAndSerializer = new HadoopUpdateStreamSorterAndSerializer(conf); 
updateSorterAndSerializer.run(personStreamsFileNames, "person");
```

I'm confused why it happened. why the intermediate files "/temp_updateStream_forum_" + i + "_" + j" were not generated successfully?

**Comment by ArnauPrat** [comment]
Are you sure you don't have a previous error that made these files not to be generated? Also, please paste your params.ini file and run.sh file.
Thanks

**Comment by yaobaiwei** [comment]
Yes, this is the first place that reported error. I directly used the default params.ini and run.sh without any changes. But I met the same error as Issue #23 said, so i fixed it by using the same approach, that is, to rename the params.ini in the root folder to params-gen.ini and modified the last parameter in run.sh to params-gen.ini accordingly.

**Comment by ArnauPrat** [comment]
Great, we will try to fix issue #23 so this does not happen again.

**Comment by szarnyasg** [comment]
#23 was fixed which caused this issue - I'm closing this as well.
